schema.go

```diff
@@ -349,10 +355,29 @@ type structNode struct {
 	fields []structField
 }
 
-func structNodeOf(t reflect.Type) *structNode {
+type tagSource interface {
+	// Tags extracts parquet-go tags from a struct field. The returned array
+	// indices contains
+	//	0: parquet tag
+	// 	1: parquet-key tag (This is for map keys)
+	// 	2: parquet-value tag (This is for map values)
+	Tags(f *reflect.StructField) [3]string
```

Comment:
The type is generally referenced by value in the `reflect` package; let's pass it by value here as well for consistency.

Reply:
done 1452723